contracts/scripts/deployRound.ts

```diff
@@ -0,0 +1,144 @@
+      'BrightIdUserRegistry',
+      deployer
+    )
+    // TODO sort out if these arguments are sensible
+    userRegistry = await BrightIdUserRegistry.deploy(
+      '0x636c722e66756e64000000000000000000000000000000000000000000000000', // Do we have a test `context`?
+      fundingRoundFactory.address // Not sure if `verifier` should be EOA or contract account
+    )
+    await userRegistry.deployTransaction.wait()
+  } else {
+    throw new Error('unsupported user registry type')
+  }
+  const setUserRegistryTx = await fundingRoundFactory.setUserRegistry(
+    userRegistry.address
+  )
+  await setUserRegistryTx.wait()
+
+  const recipientRegistryType = process.env.RECIPIENT_REGISTRY_TYPE || 'simple'
+  let recipientRegistry: Contract
+  if (recipientRegistryType === 'simple') {
+    const SimpleRecipientRegistry = await ethers.getContractFactory(
+      'SimpleRecipientRegistry',
+      deployer
+    )
+    recipientRegistry = await SimpleRecipientRegistry.deploy(
+      fundingRoundFactory.address
+    )
+    await recipientRegistry.deployTransaction.wait()
+  } else if (recipientRegistryType === 'optimistic') {
+    const OptimisticRecipientRegistry = await ethers.getContractFactory(
+      'OptimisticRecipientRegistry',
+      deployer
+    )
+    recipientRegistry = await OptimisticRecipientRegistry.deploy(
+      ethers.BigNumber.from(10).pow(ethers.BigNumber.from(18)).div(10),
+      300,
+      fundingRoundFactory.address
+    )
+    await recipientRegistry.deployTransaction.wait()
+  } else {
+    throw new Error('unsupported recipient registry type')
+  }
+
+  const setRecipientRegistryTx = await fundingRoundFactory.setRecipientRegistry(
+    recipientRegistry.address
+  )
+  await setRecipientRegistryTx.wait()
+
+  const setTokenTx = await fundingRoundFactory.setToken(
+    process.env.NATIVE_TOKEN_ADDRESS
+  )
+  await setTokenTx.wait()
+
+  // Generate coordinator key
+  const keypair = new Keypair()
+  const coordinatorPubKey = keypair.pubKey
+  const serializedCoordinatorPrivKey = keypair.privKey.serialize()
+  const serializedCoordinatorPubKey = keypair.pubKey.serialize()
+  const setCoordinatorTx = await fundingRoundFactory.setCoordinator(
+    deployer.address,
+    coordinatorPubKey.asContractParam()
+  )
+  await setCoordinatorTx.wait()
+
+  maciFactory = await ethers.getContractAt('MACIFactory', maciFactory.address)
+  const maciParameters = await MaciParameters.read(maciFactory)
+  maciParameters.update({
+    signUpDuration: 86400 * 14,
+    votingDuration: 86400 * 3,
+  })
+  const setMaciParametersTx = await fundingRoundFactory.setMaciParameters(
+    ...maciParameters.values()
+  )
+  await setMaciParametersTx.wait()
+
+  const addFundingSourceTx = await fundingRoundFactory.addFundingSource(
+    deployer.address
+  )
+  await addFundingSourceTx.wait()
+
+  const deployNewRoundTx = await fundingRoundFactory.deployNewRound()
+  await deployNewRoundTx.wait()
+
+  console.log('deployer.address')
```

Comment:
Also can we move these console logs to after the `wait()` for each deploy? That way it logs to console as the script progresses vs. leaving everything to the end.